cli/crates/common/src/environment.rs

```diff
@@ -330,6 +334,104 @@ fn find_grafbase_configuration(path: &Path, warnings: &mut Vec<Warning>) -> Opti
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub enum NodePackageManager {
+    Npm,
+    Pnpm,
+    Yarn,
+}
+
+impl Default for NodePackageManager {
+    fn default() -> Self {
+        Self::Npm
+    }
+}
+
+impl NodePackageManager {
+    #[must_use]
+    pub fn is_npm(self) -> bool {
+        matches!(self, Self::Npm)
+    }
+}
+
+impl fmt::Display for NodePackageManager {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Npm => f.write_str("npm"),
+            Self::Pnpm => f.write_str("pnpm"),
+            Self::Yarn => f.write_str("yarn"),
+        }
+    }
+}
+
+impl FromStr for NodePackageManager {
+    type Err = CommonError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "npm" => Ok(NodePackageManager::Npm),
+            "pnpm" => Ok(NodePackageManager::Pnpm),
+            "yarn" => Ok(NodePackageManager::Yarn),
+            _ => {
+                let inner = io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    format!("Unsupported package manager in package.json: {s}"),
+                );
+
+                Err(CommonError::AccessPackageJson(inner))
+            }
+        }
+    }
+}
+
+pub fn install_dependencies(project_root: &Path) -> Result<(), CommonError> {
+    let command = match which::which("npm") {
+        Ok(command) => command,
+        Err(e) => return Err(CommonError::NpmNotFound(Box::new(e))),
+    };
+
+    Command::new(command)
+        .args(["install", "--save-dev"])
+        .current_dir(project_root)
+        .output()
+        .map_err(CommonError::NpmInstall)?;
+
+    Ok(())
+}
+
+pub fn guess_project_package_manager(project_root: &Path, package_json: &Path) -> Option<NodePackageManager> {
+    const LOCK_FILES: &[(&str, NodePackageManager)] = &[
+        (PNPM_LOCK_FILE, NodePackageManager::Pnpm),
```

Comment:
I saw you did this based on @yoav-lavi's input, but, honestly, I don't see what the improvement here is. The whole slice was a const already.

Also, I'd rename this to `LOCK_FILE_NAMES`.

Reply:
I think this was in regards to a specific string rather than the whole computation, but we could put it in Project

Reply:
Except, well, `LOCK_FILE_NAMES` will be a better name